pkg/manager/namespace/manager.go

```diff
@@ -33,7 +34,7 @@ func (mgr *NamespaceManager) buildNamespace(cfg *config.Namespace) (*Namespace,
 	logger := mgr.logger.With(zap.String("namespace", cfg.Namespace))
 
 	var fetcher router.BackendFetcher
-	if mgr.tpFetcher != nil {
+	if !reflect.ValueOf(mgr.tpFetcher).IsNil() {
```

Comment:
Why?

Reply:
Even if `pdFetcher` is nil, but when it is passing to a interface, the interface becomes a non-nil interface with nil value.

I mean `mgr.tpFetcher != nil` is always true.